generator/src/main/java/com/linkedin/pegasus/generator/TemplateSpecGenerator.java

```diff
@@ -635,7 +624,8 @@ private ClassTemplateSpec generateUnion(UnionDataSchema schema, TyperefDataSchem
     pushCurrentLocation(_schemaResolver.nameToDataSchemaLocations().get(typerefDataSchema.getFullName()));
 
     final UnionTemplateSpec unionClass = new UnionTemplateSpec(schema);
-    unionClass.setNamespace(typerefDataSchema.getNamespace());
+    String namespace = _lowercaseNamespace ? typerefDataSchema.getNamespace().toLowerCase() : typerefDataSchema.getNamespace();
+    unionClass.setNamespace(namespace);
```

Comment:
This is changing the namespace on the generate template classes and it is a breaking change. This config should only change the generated directories